openassessment/xblock/staff_area_mixin.py

```diff
@@ -226,7 +235,77 @@ def render_student_info(self, data, suffix=''):  # pylint: disable=W0613
             return self.render_assessment(path, context)
 
         except PeerAssessmentInternalError:
-            return self.render_error(self._(u"Error finding assessment workflow cancellation."))
+            return self.render_error(self._(u"Error getting learner information."))
+
+    @XBlock.handler
+    @require_course_staff("STUDENT_GRADE")
+    def render_staff_grade_form(self, data, suffix=''):  # pylint: disable=W0613
+        """
+        Renders a form to staff-grade the next available learner submission.
+
+        Must be course staff to render this view.
+        """
+        try:
+            student_item_dict = self.get_student_item_dict()
+            course_id = student_item_dict.get('course_id')
+            item_id = student_item_dict.get('item_id')
+            staff_id = student_item_dict['student_id']
+
+            # Note that this will check out a submission for grading by the specified staff member.
+            # If no submissions are available for graidng, will reutrn None.
+            submission_to_assess = staff_api.get_submission_to_assess(course_id, item_id, staff_id)
+
+            if submission_to_assess is not None:
+                submission = submission_api.get_submission_and_student(submission_to_assess['uuid'])
+                if submission:
+                    anonymous_student_id = submission['student_item']['student_id']
+                    submission_context = self.get_student_submission_context(
+                        self.get_username(anonymous_student_id), submission
+                    )
+                    path = 'openassessmentblock/staff_area/oa_staff_grade_learners_assessment.html'
+                    return self.render_assessment(path, submission_context)
+                else:
+                    return self.render_error(self._(u"Error loading the checked out learner response."))
+            else:
+                return self.render_error(self._(u"No more learner responses can be graded at this time."))
```

Comment:
If the staff member expands the Staff Assessment form and no assessments are available to be graded, this is the message that will be shown. It's shown in grey text-- not a scary error sort of rendering.

No assessments can be available either because they are all graded, or because there are no ungraded assessments that are not already locked (checked out) by another staff member. When we show the number of ungraded and in-progress assessments, we can't clarify for checked-out whether or not the current staff member has one of those checked out.

As a further point of potential confusion, the number of ungraded and checked out assignments are rendered before the staff member tries to expand the Staff Assessment form. Therefore, it is possible that the staff member will see 1 (or some other small number) of ungraded assessments, but when trying to get the next assessment to grade, none may be available (if other staff grading at the same time).

Given all of that, we may wish to make a lengthier message, along the lines of "No other learner responses are available for grading at this time, either due to other staff members having them checked out, or due to other staff members grading concurrently. Refresh the page to get an updated count of ungraded and checked out assignments." (uck)

Note that we DO update counts on "submit" of an assessment (for both "submit" buttons).